src/app/SharpRaven/RavenClient.cs

```diff
@@ -325,14 +325,19 @@ public string CaptureEvent(Exception e, Dictionary<string, string> tags)
         /// <returns>The <see cref="JsonPacket"/> which should be sent to Sentry.</returns>
         protected internal virtual JsonPacket PreparePacket(JsonPacket packet)
         {
-            packet.Logger = string.IsNullOrWhiteSpace(packet.Logger)
-                            || (packet.Logger == "root" && !string.IsNullOrWhiteSpace(Logger))
+            packet.Logger = SystemUtil.IsNullOrWhiteSpace(packet.Logger)
+                            || (packet.Logger == "root" && !SystemUtil.IsNullOrWhiteSpace(Logger))
                 ? Logger
                 : packet.Logger;
             packet.User = packet.User ?? this.sentryUserFactory.Create();
-            packet.Request = packet.Request ?? this.sentryRequestFactory.Create();
-            packet.Release = string.IsNullOrWhiteSpace(packet.Release) ? Release : packet.Release;
-            packet.Environment = string.IsNullOrWhiteSpace(packet.Environment) ? Environment : packet.Environment;
+            try {
+                packet.Request = packet.Request ?? this.sentryRequestFactory.Create();
+            }
+            catch {
+                packet.Request = this.sentryRequestFactory.Create();
```

Comment:
If the above `sentryRequestFactory.Create()` failed, won't this fail as well? I don't understand this `try/catch`.